### ISSUE TITLE
Extending modules

### ISSUE THREAD

**Issue by smarlowucf** [philpep/testinfra#233]
This is more of a question than an issue, but is there a way to extend modules locally? I'm attempting to extend the File module but I don't see any documentation covering this.

There is an [older issue](https://github.com/philpep/testinfra/issues/181) which mentions using `as_fixture()`. This however appears to have been removed in recent versions.

**Comment by philpep** [comment]
Yes this definitively need some stable API and documentation to create new modules. I'm not sure if we should allow extending an existing module, what's your usecase ?

**Comment by smarlowucf** [comment]
For this case it is for testing repositories. This is basically a more specific file check.

- Check if file exists for repo
- Check in file for enabled/disabled
- Other checks in file if necessary

So being able to extend the File module would be nice to not duplicate code.

**Comment by smarlowucf** [comment]
For now I was just using `File` in a fixture but it's not as extensible as being able to create a module class.
